h5netcdf/core.py

```diff
@@ -112,6 +113,9 @@ def __init__(self, parent, name, dimensions=None):
         self._root_ref = weakref.ref(parent._root)
         self._h5path = _join_h5paths(parent.name, name)
         self._dimensions = dimensions
+        self._h5ds = self._root._h5file[self._h5path]
+        # fix name if _nc4_non_coord_
+        self._name = self._h5path.replace("_nc4_non_coord_", "")
```

Comment:
Your previous comment stated:

> dataset/name caching can be done when opened for read-only in any case, for write/append we would need to think how to apply caching/refresh cache on write

I'm not sure I know how to recreate this issue.